### PR TITLE
LPS-182815 Add DateTime object field

### PR DESCRIPTION
**Pull request by devpedrotavares**
Follow up on: https://github.com/liferay-objects/liferay-portal/pull/2072

### CODE REVIEW

**Comment by devpedrotavares** [comment]
ci:test:sf

**Comment by devpedrotavares** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
@devpedrotavares @marcelabc There are few unresolved comments left from the previous review (https://github.com/liferay-objects/liferay-portal/pull/2072), but the only one that worries me is this one [here](https://github.com/liferay - objects/liferay-portal/pull/2072#discussion_r1188731573).

**Comment by marcelabc** [comment]
Just started reviewing :)

**Comment by devpedrotavares** [comment]
Hey @marcelabc @carolmariaabb ! I did added the changes requested [here](https://github.com/liferay-objects/liferay-portal/pull/2072#discussion_r1188723794) and [here](https://github.com/liferay-objects/liferay-portal/pull/2072#discussion_r1188858787). 
I believe the last one needs review.

**Comment by devpedrotavares** [comment]
ci:test:sf

**Comment by devpedrotavares** [comment]
ci:test:object

**Comment by devpedrotavares** [comment]
Hey @marcelabc @carolmariaabb ! I added the changes requested [here](https://github.com/liferay-objects/liferay-portal/pull/2072#discussion_r1188723794) and [here](https://github.com/liferay-objects/liferay-portal/pull/2072#discussion_r1188858787). 
I believe the last one needs review: 4a9db5208ee7442d265fd06856b5e49de3f75756

I also did this [change](https://github.com/liferay-objects/liferay-portal/compare/4c281c08cb8d5a36a836d2f539b3d99fac9a03ff..ce00dd836a01dfc0940c335a1994d746aed623a2) to fix an issue with Custom Views.

**Comment by devpedrotavares** [comment]
ci:test:sf

**Comment by devpedrotavares** [comment]
ci:test:object

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/DateObjectFieldBusinessType.java [hunk 1]
```diff
@@ -64,6 +69,20 @@ public PropertyDefinition.PropertyType getPropertyType() {
 		return PropertyDefinition.PropertyType.DATE_TIME;
 	}
 
+	@Override
+	public Object getValueToDisplayContext(
+			ObjectField objectField, long userId, Map<String, Object> values)
+		throws PortalException {
+
+		String value = String.valueOf(values.get(objectField.getName()));
```
[the remark this answers is not in the document]
Comment (devpedrotavares) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/DateObjectFieldBusinessType.java:77]: That's better!

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/display/context/ObjectEntryDisplayContextImpl.java [hunk 2]
```diff
@@ -1139,37 +1138,6 @@ private boolean _isActive(ObjectField objectField) throws PortalException {
...
-		List<DDMFormField> ddmFormFields, Map<String, Object> values) {
-
-		for (DDMFormField ddmFormField : ddmFormFields) {
-			if (StringUtil.equals(ddmFormField.getType(), "date")) {
-				_removeTimeFromDateString(ddmFormField, values);
-			}
-			else if (StringUtil.equals(ddmFormField.getType(), "fieldset")) {
-				_setDateDDMFormFieldValue(
-					ddmFormField.getNestedDDMFormFields(), values);
-			}
```
[the remark this answers is not in the document]
Comment (devpedrotavares) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/display/context/ObjectEntryDisplayContextImpl.java:1169]: I tried it: 


https://github.com/liferay-objects/liferay-portal/assets/82452840/45d1c632-214a-4f5e-99e2-67c6060960be

**Comment by devpedrotavares** [comment]
ci:test:sf

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java [hunk 3]
```diff
@@ -148,6 +148,14 @@ public static com.liferay.object.model.ObjectField toObjectField(
 		ObjectFieldSettingLocalService objectFieldSettingLocalService,
 		ObjectFilterLocalService objectFilterLocalService) {
 
+		if (!FeatureFlagManagerUtil.isEnabled("LPS-143068") &&
```
[the remark this answers is not in the document]
Comment (devpedrotavares) [reply modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java:151]: You're right! the PUT scenario was not being covered. To fix it, I added the FF [here](https://github.com/liferay-objects/liferay-portal/pull/2089/files#diff-05788445836e16c530386e55062a7c76a71e8ce4d60428c58cee9b01a1b50966R509) as well

File: modules/dxp/apps/osb/osb-faro/.gitrepo [hunk 4]
```diff
@@ -4,8 +4,8 @@
 [subrepo]
 	autopull = false
 	cmdver = liferay
-	commit = 64b4f23cdd7b9366ba9d9ceac9288357dac22242
```
[the remark this answers is not in the document]
Comment (devpedrotavares) [reply modules/dxp/apps/osb/osb-faro/.gitrepo]: This was added by the bot

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java [hunk 5]
```diff
@@ -688,7 +689,8 @@ private Object _getValue(
 					objectField.getBusinessType());
 
 			Object value = objectFieldBusinessType.getValue(
-				objectField, Collections.singletonMap(entityFieldName, right));
+				objectField, PrincipalThreadLocal.getUserId(),
```
[the remark this answers is not in the document]
Comment (devpedrotavares) [reply modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java:692]: I tested it and the `PrincipalThreadLocal` does have the `userId` at this point. However, the filter using dateTime field was not working. To fix it, I did [this](https://github.com/liferay-objects/liferay-portal/pull/2089/files#diff-067ea9ec20926a534c8a036e949b1ca38fc8a116367916b55e258cbc23b4efadR119)

**Comment by devpedrotavares** [comment]
ci:test:object

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by devpedrotavares** [comment]
ci:test:sf

**Comment by devpedrotavares** [comment]
ci:test:object

**Comment by devpedrotavares** [comment]
ci:test:relevant

**Comment by devpedrotavares** [comment]
ci:test:sf

**Comment by devpedrotavares** [comment]
ci:test:relevant

**Comment by devpedrotavares** [comment]
ci:test:object

**Review by markocikos — APPROVED** [review]
FDS part LGTM! :sparkles:

**Comment by marcelabc** [comment]
Thank you, @markocikos !

**Comment by marcelabc** [comment]
resend here: https://github.com/liferay-objects/liferay-portal/pull/2112

**Pull request closed without merging**